plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,548 @@
+                required_one_of=[
+                    ["scope", "scope_uuid"],
+                    ["ipsla_monitoring_policy", "ipsla_monitoring_policy_uuid"],
+                ],
+                options=dict(
+                    destination_ip=dict(type="str", aliases=["ip"], required=True),
+                    ipsla_monitoring_policy=dict(
+                        type="dict",
+                        options=dict(
+                            name=dict(type="str", required=True),
+                        ),
+                    ),
+                    ipsla_monitoring_policy_uuid=dict(type="str"),
+                    scope_uuid=dict(type="str"),
+                    scope=dict(
+                        type="dict",
+                        options=dict(
+                            name=dict(type="str", required=True),
+                            template=dict(type="str"),
+                            template_id=dict(type="str"),
+                            schema=dict(type="str"),
+                        ),
+                        required_one_of=[
+                            ["template", "template_id"],
+                        ],
+                        mutually_exclusive=[
+                            ("template", "template_id"),
+                        ],
+                    ),
+                    scope_type=dict(type="str", choices=["bd", "l3out"], required=True),
+                    weight=dict(type="int"),
+                ),
+            ),
+            state=dict(type="str", choices=["absent", "query", "present"], default="present"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ("template", "template_id"),
+        ],
+        required_if=[
+            ["state", "absent", ["ipsla_track_list", "ipsla_track_list_uuid"], True],
+            ["state", "present", ["ipsla_track_list", "ipsla_track_list_uuid"], True],
+            ["state", "present", ["type"]],
+        ],
+        required_one_of=[
+            ["template", "template_id"],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    template_id = module.params.get("template_id")
+    ipsla_track_list = module.params.get("ipsla_track_list")
+    description = module.params.get("description")
+    ipsla_track_list_uuid = module.params.get("ipsla_track_list_uuid")
+    list_type = module.params.get("type")
+    thresholds = {
+        "down": module.params.get("threshold_down"),
+        "up": module.params.get("threshold_up"),
+    }
+    members = module.params.get("members")
+    state = module.params.get("state")
+
+    # Validate
+    valid_upper = 100
+    valid_lower = 0
+    if list_type == "weight":
+        valid_upper = 255
+    for threshold_key, threshold_value in thresholds.items():
+        if threshold_value is not None and threshold_value not in range(valid_lower, valid_upper):
```

Comment:
should we append all the potential errors (up and down) in threshold to fail in single error?

Reply:
There are only 2 thresholds in the dictionary, up and down. We don't usually append all validation errors into one message.
If the validation was in a list of items it would be useful but I think because its only two options this should be okay.